students/eldonab/lesson04/mailroom2.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+from pathlib import Path
+
+#create a list of at least 5 donors.
+#updated donors_list from part one using a dictionary
+donors_list = {
+"Bill Gates": [50000, 600000],
+"Jeff Bezos": [100000, 400, 7000],
+"Hannah Smith": [400000, 600000], 
+"John Clark": [30000, 40000],
+"Andrew Jones": [80000,90000, 1000]
+}
+
+#updated all functions using dictionary methods:
+
+def add_donation_to_list(res_name):
+    """Add a new donation amount to an existing donor's name."""
+    amount = float(input("How much is the new donation amount? "))
+    for donor in donors_list:
+        if res_name in donor:
+            donors_list[donor].append(amount)
+            letter(res_name, amount)
```

Comment:
couple comments here:
* you want to avoid referring to your donor db as "list" since it's no longer a list and can confuse the reader (good variables names are important as code is read more than it is written)
* no need to loop over dict keys to check if donor is there (I will have an example below)
* to reduce duplication, combine this and `add_new_donor` functions together.


suggestion:
```
def add_or_update_donor(name):
    amount = input("") 
    if name in donors:
        # use .append() here
    else:
        # assign list to new key
    letters(name, amount)
```